airflow/dag_processing/bundles/git.py

```diff
@@ -68,10 +68,10 @@ def get_ui_field_behaviour(cls) -> dict[str, Any]:
             },
         }
 
-    def __init__(self, git_conn_id="git_default", *args, **kwargs):
+    def __init__(self, git_conn_id: str = "git_default", repo_url: str | None = None, *args, **kwargs) -> None:
```

Comment:
Do we need the repo_url on the hook or just take it from the connection host? Should we remove the repo_url on the GitDagBundle itself?

Reply:
We added it back intentionally: https://github.com/apache/airflow/issues/45952

For cases where its either a public repo or if keys are already configured on your system, you can operate without a connection.

Reply:
It also provides flexibility where the connection can hold a shared credential and the repo to use is set in the bundle itself.